service/service.go

```diff
@@ -279,168 +328,145 @@ func (s *Service) Started() bool {
 	return s.started.Load()
 }
 
-func (s *Service) recover(ctx context.Context) error {
-	log.With().Info("Recovering service state", log.String("datadir", s.datadir))
+func (s *Service) recover(ctx context.Context) (open *round, executing []*round, err error) {
+	logger := log.AppLog.WithName("recovery")
+	logger.With().Info("Recovering service state", log.String("datadir", s.datadir))
 	entries, err := os.ReadDir(s.datadir)
```

Comment:
This seems like a good place to use https://pkg.go.dev/path/filepath#WalkDir. What do you think?

Reply:
Great idea. I will explore it in a separate PR refactoring `Service::recover()`. I created an issue to track it https://github.com/spacemeshos/poet/issues/179